packages/smooth_app/lib/pages/user_management/login_page.dart

```diff
@@ -0,0 +1,240 @@
+              children: <Widget>[
+                const Spacer(flex: 4),
+
+                //Text
+                Text(
+                  'Sign in to your Open Food Facts account to save your contributions ',
+                  textAlign: TextAlign.center,
+                  style: theme.textTheme.headline1?.copyWith(
+                    fontSize: 20.0,
+                    fontWeight: FontWeight.w700,
+                    color:
+                        themeProvider.darkTheme ? Colors.white : Colors.black,
+                  ),
+                ),
+
+                const Spacer(flex: 8),
+
+                if (_wrongCredentials) ...const <Widget>[
+                  SmoothCard(
+                    padding: EdgeInsets.all(10.0),
+                    color: Colors.red,
+                    child: Text('Incorrect user name or password.'),
+                  ),
+                  Spacer(
+                    flex: 1,
+                  )
+                ],
+
+                //Login
+                SmoothTextFormField(
+                  type: TextFieldTypes.PLAIN_TEXT,
+                  controller: userIdController,
+                  hintText: 'Login',
+                  textColor: customGrey,
+                  backgroundColor: textFieldBackgroundColor,
+                  prefixIcon: const Icon(Icons.person),
+                  enabled: !_runningQuery,
+                  textInputAction: TextInputAction.next, // Moves focus to next.
+                  validator: (String? value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Please enter some text';
+                    }
+                    return null;
+                  },
+                ),
+
+                const Spacer(flex: 1),
+
+                //Password
+                SmoothTextFormField(
+                  type: TextFieldTypes.PASSWORD,
+                  controller: passwordController,
+                  hintText: 'Password',
+                  textColor: customGrey,
+                  backgroundColor: textFieldBackgroundColor,
+                  prefixIcon: const Icon(Icons.vpn_key),
+                  enabled: !_runningQuery,
+                  textInputAction: TextInputAction.done, // Hides the keyboard
+                  validator: (String? value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Please enter some text';
+                    }
+                    return null;
+                  },
+                ),
+
+                const Spacer(flex: 6),
+
+                //Sign in button
+                ElevatedButton(
+                  onPressed: () => _login(context),
+                  child: Text(
+                    'Sign in',
+                    style: theme.textTheme.bodyText2?.copyWith(
+                      fontSize: 18.0,
+                      color:
+                          themeProvider.darkTheme ? Colors.black : Colors.white,
```

Comment:
There's probably a color in `SmoothTheme` for that AND you should create a local `final`.

Reply:
Yes there was on, what you mean by final, probably a final bool darkmode but now we don't need this anymore do we